### PR TITLE
common: fs: Use the normal directory iterator in *Recursively functions

### PR DESCRIPTION
**Pull request by Morph1984**
MSVC's implementation of recursive_directory_iterator throws an exception on an error despite a std::error_code being passed into its constructor. This is most likely a bug in MSVC's implementation since directory_iterator does not throw an exception on an error.

We can replace the usage of recursive_directory_iterator for now until MSVC fixes their implementation of it.

This fixes crashes when trying to access folders without sufficient permissions (such as `System Volume Information` at the root of a drive).

### CODE REVIEW

**Review by FernandoS27 — APPROVED** [review]

**Comment by CookiePLMonster** [comment]
Calling this a MSVC bug is not correct, it was an implementation bug (with this PR fixing the bug correctly).

`recursive_directory_iterator` appears to build the directory tree eagerly, not lazily; therefore, the code removing directories recursively creates a bug where advancing the iterator attempts to recurse to a now removed directory. While I agree this should have set an error code, a case like this does not seem to be explicitly covered in the documentation:
https://en.cppreference.com/w/cpp/filesystem/recursive_directory_iterator/increment
and therefore it probably can do whatever.

TL;DR: The fix in PR is correct but notes about a "MSVC bug" are not.
